test/orchard/inspect_test.clj

```diff
@@ -319,3 +368,143 @@
+               (:newline)
+               "Interns"
+               ": " (:value "{}" 3) (:newline) "---" (:newline)
+               "Datafy: "
+               (:newline)
+               "Class"
+               ": " (:value "clojure.lang.PersistentArrayMap" 4) (:newline)
+               "Contents: "
+               (:newline) "  " (:value ":name" 5) " = " (:value "orchard.inspect-test-ns" 6)
+               (:newline) "  " (:value ":publics" 7) " = " (:value "{}" 8)
+               (:newline) "  " (:value ":imports" 9) " = " (:value "{ AbstractMethodError java.lang.AbstractMethodError, Appendable java.lang.Appendable, ArithmeticException java.lang.ArithmeticException, ArrayIndexOutOfBoundsException java.lang.ArrayIndexOutOfBoundsException, ArrayStoreException java.lang.ArrayStoreException, ... }" 10)
+               (:newline) "  " (:value ":interns" 11) " = " (:value "{}" 12)
+               (:newline))
+             (-> (create-ns 'orchard.inspect-test-ns) inspect render))))))
+
+(when datafy?
+  (deftest inspect-metadata-extension-datafy-test
+    (testing "inspecting a map extended with `p/datafy` renders the datafy section"
+      (is (= '("Datafy: "
+               (:newline)
+               "Class"
+               ": "
+               (:value "clojure.lang.PersistentArrayMap" 5)
+               (:newline)
+               "Contents: "
+               (:newline) "  " (:value ":name" 6) " = " (:value "\"John Doe\"" 7)
+               (:newline) "  " (:value ":type" 8) " = " (:value "Person" 9)
+               (:newline))
+             (-> (with-meta {:name "John Doe"}
+                   {'clojure.core.protocols/datafy (fn [x] (assoc x :type 'Person))})
+                 inspect render skip-to-datafy-section))))))
+
+(when datafy?
+  (deftest inspect-nav-extension-datafy-test
+    (testing "inspecting a map extended with `p/nav` renders the datafy section"
+      (is (= '("Datafy: "
+               (:newline) "Class" ": " (:value "clojure.lang.PersistentArrayMap" 7) (:newline)
+               "Contents: "
+               (:newline) "  " (:value ":name" 8) " = " (:value "[ :name \"John Doe\" ]" 9)
+               (:newline) "  " (:value ":type" 10) " = " (:value "[ :type Person ]" 11)
+               (:newline))
+             (-> (with-meta {:name "John Doe"}
+                   {'clojure.core.protocols/datafy
+                    (fn [x] (assoc x :type 'Person))
+                    'clojure.core.protocols/nav
+                    (fn [coll k v]
+                      [k (get coll k v)])})
+                 inspect render skip-to-datafy-section))))))
+
+(when datafy?
+  (deftest inspect-throwable-datafy-test
+    (testing "inspecting a throwable renders the datafy section"
+      (let [data (-> (doto ^Throwable (ex-info "BOOM" {})
+                       (.setStackTrace (into-array StackTraceElement [])))
+                     inspect render skip-to-datafy-section)]
+        ;; Differences in JDKs :/
+        (is (or (= '("Datafy: "
```

Comment:
Let's code it approximately like this instead, leaving something with fixed expectations:

```clj
(case misc/java-api-version
  8 (is (= ,,,))
  (11 16 17) (is (= ,,,))
  (assert false "Not clause for this JDK"))
```